### PR TITLE
add RKEVersion and remove K8sVersion data

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/20368

### CODE REVIEW

File: apis/management.cattle.io/v3/rke_types.go [hunk 1]
```diff
@@ -691,7 +728,7 @@ type RotateCertificates struct {
 
 type DNSConfig struct {
 	// DNS provider
-	Provider string `yaml:"provider" json:"provider,omitempty"`
+	Provider string `yaml:"provider" json:"provider,omitempty" norman:"default=kube-dns"`
```
Comment (alena1108) [inline apis/management.cattle.io/v3/rke_types.go]: you should rebase with master, so you get this change: https://github.com/rancher/types/pull/865/files

File: apis/management.cattle.io/v3/tools_system_images.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		LoggingSystemImages: LoggingSystemImages{
+			Fluentd:                       m("rancher/fluentd:v0.1.11"),
+			FluentdHelper:                 m("rancher/fluentd-helper:v0.1.2"),
+			LogAggregatorFlexVolumeDriver: m("rancher/log-aggregator:v0.1.4"),
+		},
+		AuthSystemImages: AuthSystemImages{
+			KubeAPIAuth: m("rancher/kube-api-auth:v0.1.3"),
+		},
+	}
+)
```
Comment (prachidamle) [inline apis/management.cattle.io/v3/tools_system_images.go]: Do these stay same for all k8s versions?

File: apis/management.cattle.io/v3/rke_types.go [hunk 3]
```diff
@@ -169,6 +174,48 @@ type RKEConfigNode struct {
...
+	Template string `yaml:"template" json:"template,omitempty"`
+}
+
+type RKEK8sWindowsSystemImage struct {
+	types.Namespaced
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	SystemImages WindowsSystemImages `yaml:"windows_system_images" json:"windowsSystemImages,omitempty"`
+}
```
Comment (prachidamle) [inline apis/management.cattle.io/v3/rke_types.go]: Is there a reason to have these types namespaced? do they belong to cattle-global-data?

Comment (kinarashah) [reply apis/management.cattle.io/v3/rke_types.go]: Yeah, they belong to cattle-global-data

File: apis/management.cattle.io/v3/tools_system_images.go (hunk 2 above)
Comment (kinarashah) [reply apis/management.cattle.io/v3/tools_system_images.go]: Yeah, toolsSystemImages don't change with k8s version

**Pull request merged**